extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java

```diff
@@ -55,72 +56,73 @@ public void startVaultContainers(BuildProducer<DevServicesConfigResultBuildItem>
 
         // figure out if we need to shut down and restart any existing Vault container
         // if not and the Vault container have already started we just return
-        if (closeables != null) {
+        if (closeable != null) {
             boolean restartRequired = !currentDevServicesConfiguration.equals(capturedDevServicesConfiguration);
             if (!restartRequired) {
                 return;
             }
-            for (Closeable closeable : closeables) {
-                try {
-                    closeable.close();
-                } catch (Throwable e) {
-                    log.error("Failed to stop Vault container", e);
-                }
+            try {
+                closeable.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop Vault container", e);
             }
-            closeables = null;
             capturedDevServicesConfiguration = null;
         }
 
         capturedDevServicesConfiguration = currentDevServicesConfiguration;
 
-        StartResult startResult;
+        VaultInstance vaultInstance;
```

Comment:
why keeping the declaration out of the `try`?

Reply:
Fixed.